src/Lynx/List.purs

```diff
@@ -0,0 +1,216 @@
+  , Output
+  , View
+  , component
+  , fromJSON
+  , toJSON
+  ) where
+
+import Prelude
+import Data.Argonaut as Data.Argonaut
+import Data.Codec.Argonaut as Data.Codec.Argonaut
+import Data.Either (Either(..))
+import Data.Maybe (Maybe(..))
+import Data.Symbol (SProxy(..))
+import Data.Traversable as Data.Traversable
+import Halogen as Halogen
+import Halogen.HTML as Halogen.HTML
+import Lynx.Expr as Lynx.Expr
+import Ocelot.Block.Card as Ocelot.Block.Card
+import Ocelot.Block.Format as Ocelot.Block.Format
+import Ocelot.Block.Table as Ocelot.Block.Table
+import Ocelot.HTML.Properties as Ocelot.HTML.Properties
+
+-- | The intent of providing `JSONView` as an abstraction instead of using doing
+-- | our serialization directly on `View` is that it decouples the
+-- | implementation of `View` from the serialization of the concept of a "view."
+-- | Hopefully, this should keep downstream consumers from having to change
+-- | in lockstep with our codebase as our ideas and understanding evolve
+-- | (E.g. renaming fields in our type).
+-- |
+-- | While it might look almost the exact same as the `View` right now,
+-- | we provide `JSONView` a separate entity so we always suppor the decoupling.
+type JSONView
+  = { columns :: Array JSONColumn
+    }
+
+type JSONColumn
+  = { name :: Lynx.Expr.Expr
+    , value :: Lynx.Expr.Key
+    }
+
+-- | The respresentation of a "view" that we really want to work with.
+-- | As our understanding of what we're implementing grows,
+-- | we will want to make changes to `View`
+-- | rather directly in the serialization (`JSONView`).
+-- |
+-- | This is provided so we don't have to be careful about our serialization.
+-- | We should not fear changing anything about `View`,
+-- | so long as it's isomorphic to `JSONView`.
+type View
+  = { columns :: Array Column
+    }
+
+type Column
+  = { name :: Lynx.Expr.Expr
+    , value :: Lynx.Expr.Key
+    }
+
+-- | The representation of a "view" that we want to render.
+-- | This representation is also different from the `View` so we can make
+-- | changes to how we render things without forcing those changes to ripple
+-- | back into the logic of what a "view" really does.
+type RenderView
+  = { columns :: Array RenderColumn
+    }
+
+type RenderColumn
+  = { name :: Lynx.Expr.ExprType
+    }
+
+type Input
+  = View
+
+data Query a
+
+type Output
+  = Void
+
+type State
+  = Either Lynx.Expr.EvalError RenderView
+
+component :: forall f. Halogen.Component Halogen.HTML.HTML Query Input Output f
+component =
+  Halogen.component
+    { eval
+    , initialState
+    , receiver
+    , render
+    }
+
+eval :: forall f. Query ~> f
+eval query = eval query
+
+fromArgonaut ::
+  forall a.
+  Data.Argonaut.DecodeJson a =>
+  Data.Argonaut.EncodeJson a =>
+  Data.Codec.Argonaut.JsonCodec a
+fromArgonaut = Data.Codec.Argonaut.prismaticCodec from to Data.Codec.Argonaut.json
+  where
+  from :: Data.Argonaut.Json -> Maybe a
+  from json = case Data.Argonaut.decodeJson json of
```

Comment:
```purescript
from = Data.Either.hush <<< Data.Argonaut.decodeJson
```
?

Reply:
Or do we prefer the explicit casing?

Reply:
I prefer casing, but I'm down to switch.

Reply:
Ah it's no big deal!